src/main/java/com/google/sps/servlets/GetVendorServlet.java

```diff
@@ -0,0 +1,105 @@
+    String vendorId = HttpServletUtils.getParameter(request, "vendorId", "");
+    float latitude, longitude;
+    GeoPt clientLocation;
+    try {
+      // If not provided, we set them to 360 to throw an error when trying to use them to create a GeoPt
+      latitude = Float.parseFloat(HttpServletUtils.getParameter(request, "lat", "360"));
+      longitude = Float.parseFloat(HttpServletUtils.getParameter(request, "long", "360"));
+      clientLocation = new GeoPt(latitude, longitude);
+    } catch (NumberFormatException e) {
+      System.out.println("The string is not a parsable float: " + e);
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Wrong format of parameters");
+      return;
+    } catch (IllegalArgumentException e) {
+      System.out.println(e);
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Latitude and/or longitude outside range");
+      return;
+    }
+
+    if (vendorId.isEmpty()) {
+      System.out.println("No vendor provided");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "No vendor provided.");
+      return;
+    }
+
+    Entity vendorEntity = HttpServletUtils.getVendorEntity(vendorId);
+    if (vendorEntity == null) {
+      System.out.println("Vendor Account does not exist");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Vendor Account does not exist");
+      return;
+    }
+
+    Vendor vendorObject = new Vendor(vendorEntity);
+    // Add the distance between client and vendor's business
+    if (vendorObject.getSaleCard() != null) {
+      float distanceClientVendor = HttpServletUtils.computeGeoDistance(
+          clientLocation, vendorObject.getSaleCard().getLocation().getSalePoint());
+      vendorObject.getSaleCard().setDistanceFromClient(distanceClientVendor);
+    }
+
+    response.setContentType("application/json;");
+    response.setCharacterEncoding("UTF-8");
+    response.getWriter().println(new Gson().toJson(vendorObject));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    
+    if (userService.isUserLoggedIn()) {
+      Entity vendorEntity = HttpServletUtils.getVendorEntity(userService.getCurrentUser().getUserId());
+      if (vendorEntity == null) {
+        System.out.println("Vendor Account does not exist");
+        response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Vendor Account does not exist");
+        return;
+      }
+
+      Vendor vendorObject = new Vendor(vendorEntity);
+      response.setContentType("application/json;");
+      response.setCharacterEncoding("UTF-8");
+      response.getWriter().println(new Gson().toJson(vendorObject));
+    } else {
+      System.out.println("User is not logged in.");
+      response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "User is not logged in.");
+    }
```

Comment:
Nit: To avoid more indentation, you can move this check to the top of the function and return early. Then for the rest of the function, you can assume the user is logged in.

Reply:
True, thanks for the advice!